packages/manager/src/managers/customTypes/CustomTypesManager.ts

```diff
@@ -642,18 +634,28 @@ export function updateCustomTypeContentRelationships(
 	const { models, previousPath, newPath, onUpdate } = args;
 
 	for (const customType of models) {
+		// Keep track of whether the model has changed to avoid calling the
+		// update hook if nothing has changed
+		let changed = false;
+
 		const updatedCustomTypeModel = traverseCustomType({
 			customType: customType.model,
 			onField: ({ field }) => {
-				return updateFieldContentRelationships({
+				const update = updateFieldContentRelationships({
 					field,
 					previousPath,
 					newPath,
 				});
+
+				changed ||= update.changed;
+
+				return update.field;
 			},
 		});
 
-		onUpdate(updatedCustomTypeModel);
+		if (changed) {
+			onUpdate(updatedCustomTypeModel);
```

Comment:
❓ I'm not sure to get all the indirection, when we are saving we are saving an entity. The whole entity. So why not just when we call the hook to save the custom type update we compare there. Not before not multiple times.

Reply:
Not critical, but the idea behind it was to compare the least amount of properties, but I agree that it's not critical, so I will reduce the indirection.

Reply:
Tell me what you think now

Reply:
I love it! I feel simpler to read as think. Like if it's equal we don't do anything, no need to read more.